src/main/java/com/example/springbootproject/controller/ChainController.java

```diff
@@ -56,67 +66,93 @@ List<ChainName> getAllDtoNames() {
         } catch (Exception e) {
             throw new ResponseStatusException(HttpStatus.NOT_FOUND);
         }
+
     }
 
     @PostMapping
     ResponseEntity<Void> addChain(@RequestBody Chain chain) {
         String name = chain.getName();
-        if (name == null || name.isEmpty())
+        if (name.isEmpty())
             throw new IllegalStateException();
-        repository.save(chain);
+
+        chainRepository.save(chain);
+
         URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}")
                 .buildAndExpand(chain.getId()).toUri();
         return ResponseEntity.created(location).build();
-
     }
 
     @DeleteMapping("/{id}")
     void delete(@PathVariable long id) {
-        repository.deleteById(id);
+        if (chainRepository.findById(id).isPresent())
+            chainRepository.deleteById(id);
+        else
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
     }
 
     @PutMapping("/{id}")
     ResponseEntity<Chain> updateChain(@PathVariable Long id, @RequestBody Chain chain) {
-        Chain updateChain = repository.findById(id)
+        Chain updateChain = chainRepository.findById(id)
                 .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
 
         updateChain.setName(chain.getName());
         updateChain.setAddress(chain.getAddress());
         updateChain.setMembers(chain.getMembers());
+        updateChain.setStores(chain.getStores());
 
-        repository.save(updateChain);
+        chainRepository.save(updateChain);
 
         return ResponseEntity.ok(updateChain);
     }
 
     @PutMapping("{chainId}/members/{memberId}")
     @Transactional
     public void addMemberToChain(@PathVariable Long memberId, @PathVariable Long chainId) {
-        repository.findById(chainId)
+        chainRepository.findById(chainId)
                 .ifPresent(chain -> chain.getMembers().add(memberRepo.findById(memberId)
                         .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND))));
     }
 
     @DeleteMapping("{chainId}/members/{memberId}")
     @Transactional
     public void deleteMemberFromChain(@PathVariable Long chainId, @PathVariable Long memberId) {
-        repository.findById(chainId)
+        chainRepository.findById(chainId)
                 .ifPresent(chain -> chain.getMembers().removeIf(member -> member.getId().equals(memberId)));
     }
 
-
+    @PutMapping("{chainId}/stores/{storeId}")
+    @Transactional
+    public void addStoreToChain(@PathVariable Long chainId, @PathVariable Long storeId) {
+            Store store = storeRepository.findById(storeId).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+            Chain chain = chainRepository.findById(chainId).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+            chain.getStores().add(store);
 }
+
+    @DeleteMapping("{chainId}/stores/{storeId}")
```

Comment:
```suggestion
    @DeleteMapping("/{chainId}/stores/{storeId}")
```

We need the forward slash before {chainId)